src/setup_payload/QRCodeSetupPayloadGenerator.cpp

```diff
@@ -202,16 +205,33 @@ CHIP_ERROR QRCodeSetupPayloadGenerator::payloadBase38Representation(std::string
                                                                     uint32_t tlvDataStartSize)
 {
     size_t tlvDataLengthInBytes = 0;
+    std::vector<uint8_t> bits(kTotalPayloadDataSizeInBytes);
+    MutableByteSpan bitsSpan;
+    std::vector<char> buffer;
+    MutableCharSpan bufferSpan;
 
     VerifyOrReturnError(mPayload.isValidQRCodePayload(), CHIP_ERROR_INVALID_ARGUMENT);
     ReturnErrorOnFailure(generateTLVFromOptionalData(mPayload, tlvDataStart, tlvDataStartSize, tlvDataLengthInBytes));
 
-    return payloadBase38RepresentationWithTLV(mPayload, base38Representation, kTotalPayloadDataSizeInBytes + tlvDataLengthInBytes,
-                                              tlvDataStart, tlvDataLengthInBytes);
+    bits.reserve(kTotalPayloadDataSizeInBytes + tlvDataLengthInBytes);
```

Comment:
Why not just put the `std::vector` constructor down here, where we know the right size?

Either way, we need to verify that the allocated capacity is actually large enough here, no?  The allocation can fail.  And the code that later writes to this buffer assumes it's big enough and never checks, afaict.

A followup PR to make `generateBitSet` take a MutableSpan, not a raw pointer, and check the size would be great.

Reply:
Fixed. Also updated generateBitSet with MutableSpan and a size check.